### PR TITLE
[10.6.X] Use OpenBlas instead of GSL Blas implementation

### PR DESCRIPTION
**Pull request by mrodozov**
please test
backport of #5540

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
backport of #5540

**Comment by smuzaffar** [comment]
please test

**Comment by silviodonato** [comment]
@smuzaffar shall we integrate this backport?

**Comment by smuzaffar** [comment]
lets wait for the comparison results. looks like there are too many differences

File: root.spec [hunk 1]
```diff
@@ -136,7 +138,7 @@ cmake ../%{n}-%{realversion} \
   -DLIBLZ4_LIBRARY="${LZ4_ROOT}/lib/liblz4.%{soext}" \
   -DZLIB_ROOT="${ZLIB_ROOT}" \
   -DZLIB_INCLUDE_DIR="${ZLIB_ROOT}/include" \
-  -DCMAKE_PREFIX_PATH="${XZ_ROOT};${OPENSSL_ROOT};${GIFLIB_ROOT};${FREETYPE_ROOT};${PYTHON_ROOT};${LIBPNG_ROOT};${PCRE_ROOT};${TBB_ROOT};${OPENBLAS_ROOT};${DAVIX_ROOT};${LZ4_ROOT};${LIBXML2_ROOT}"
+  -DCMAKE_PREFIX_PATH="${GSL_ROOT}:${XZ_ROOT};${OPENSSL_ROOT};${GIFLIB_ROOT};${PYTHON_ROOT};${LIBPNG_ROOT};${PCRE_ROOT};${TBB_ROOT};${OPENBLAS_ROOT};${DAVIX_ROOT};${LZ4_ROOT};${LIBXML2_ROOT}"
```
Comment (hqucms) [inline root.spec]: Is the change on this line needed? I did not see it in the other PRs.

Comment (smuzaffar) [reply root.spec]: cmake might have complaint about missing `FREETYPE_ROOT` path which is only used for OSX. I think change is harmless here.

**Comment by smuzaffar** [comment]
@mrodozov , please use gcc7 branch (comparison failures are due to the fact that we are compaing gcc8 vs gcc7 here).

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
merge

**Pull request merged**